numpy/core/src/multiarray/methods.c

```diff
@@ -2694,6 +2696,152 @@ array_complex(PyArrayObject *self, PyObject *NPY_UNUSED(args))
     return c;
 }
 
+#define NPY_DLPACK_CAPSULE_NAME "NumPy DLPack Wrapper"
+
+static void array_dlpack_capsule_deleter(PyObject *self)
+{
+    DLManagedTensor *managed = 
+        (DLManagedTensor *)PyCapsule_GetPointer(self, NPY_DLPACK_CAPSULE_NAME);
+    managed->deleter(managed);
+}
+
+static void array_dlpack_deleter(DLManagedTensor *self)
+{
+    PyArrayObject *array = (PyArrayObject *)self->manager_ctx;
+    free(self->dl_tensor.shape);
+    free(self->dl_tensor.strides);
+    free(self);
+
+    PyArray_XDECREF(array);
+}
+
+static PyObject *
+array_dlpack(PyArrayObject *self,
+        PyObject *const *args, Py_ssize_t len_args, PyObject *kwnames)
+{
+    PyObject *stream = Py_None;
+    NPY_PREPARE_ARGPARSER;
+    if (npy_parse_arguments("__dlpack__", args, len_args, kwnames,
+                             "$stream", NULL, &stream,
+                             NULL, NULL, NULL))
+    {
+        return NULL;
+    }
+
+    if (stream != Py_None)
+    {
+        PyErr_SetString(PyExc_RuntimeError, "NumPy only supports stream=None.");
+        return NULL;
+    }
```

Comment:
This might be more of a problem with the array-api docs, but [`__dlpack__`](https://data-apis.org/array-api/latest/API_specification/array_object.html#dlpack-self-stream-none) currently makes no mention of integer-support being optional for `stream`.

Reply:
Streams are irrelevant for CPU arrays, IIUC, so I make sure it's not supplied. Another decision could be to ignore it completely.

Reply:
The parameter description is: `stream : Optional[ int ]`

Anything except `None` is clearly incorrect on CPU, so +1 for raising.

Reply:
I agree that raising is the correct decision here, but the docs currently suggest that `__dlpack__` implementations _must_ support both `None` and `int` for the `stream` parameter. 
In my opinion it's worth clarifying that support for `stream: int` is, in fact, optional.

Reply:
Okay, just a "on CPU and other devices without streams, only None is accepted"?

Reply:
> Okay, just a "on CPU and other devices without streams, only None is accepted"?

Yeah, this sounds about right.